lib/blockscore/base.rb

```diff
@@ -60,8 +60,16 @@ def persisted?
       !id.nil? && !deleted?
     end
 
+    def deleted?
+      attributes.fetch(:deleted, false)
+    end
+
     private
 
+    def already_deleted
```

Comment:
I didn't like the name either. Of the two I am probably more adept at caching :). I will go with your first choice.